addons/website_sale_wishlist/static/src/js/website_sale_wishlist.js

```diff
@@ -78,15 +78,18 @@ publicWidget.registry.ProductWishlist = publicWidget.Widget.extend(VariantMixin,
         if ($el.hasClass('o_add_wishlist_dyn')) {
             productID = parseInt($el.closest('.js_product').find('.product_id:checked').val());;
         }
-        var $form = $el.closest('form');
-        var templateId = $form.find('.product_template_id').val();
+        let $form = $el.closest('form');
+        if ($form.length === 0) {
+            $form = $el.closest('.js_product').find('form');
+        }
+        let templateId = $form.find('.product_template_id').val();
```

Comment:
```suggestion
        var templateId = $form.find('.product_template_id').val();
```
This line can remain as is.